arangod/Aql/Condition.cpp

```diff
@@ -1093,7 +1093,8 @@ AstNode* Condition::removeCondition(ExecutionPlan const* plan,
 }
 
 /// @brief remove (now) invalid variables from the condition
-bool Condition::removeInvalidVariables(VarSet const& validVars) {
+bool Condition::removeInvalidVariables(VarSet const& validVars,
+                                       bool& noRemoves) {
```

Comment:
Even after staring at the whole code (not just the diff) for multiple minutes it is not clear to me what the relevance of this out-variable (?) is and what the optimizer rule's intent is.

Can you explain that to me?

Reply:
@markuspf as this is a bugfix  and we want it to be merged soon. Have you any more  comments to the PR?